src/renderer/wg_engine/tvgWgPipelines.cpp

```diff
@@ -223,33 +223,171 @@ void WgPipelineImage::initialize(WGPUDevice device)
+    auto pipelineLabel = "The render pipeline compose alpha mask";
+
+    // allocate all pipeline handles
+    allocate(device,
+             vertexBufferLayouts, ARRAY_ELEMENTS_COUNT(vertexBufferLayouts),
+             bindGroupLayouts, ARRAY_ELEMENTS_COUNT(bindGroupLayouts),
+             stencilFuncion, stencilOperation,
+             shaderSource, shaderLabel, pipelineLabel);
+}
+
 //************************************************************************
 // pipelines
 //************************************************************************
 
 void WgPipelines::initialize(WGPUDevice device)
 {
-    mPipelineFillShape.initialize(device);
-    mPipelineFillStroke.initialize(device);
-    mPipelineSolid.initialize(device);
-    mPipelineLinear.initialize(device);
-    mPipelineRadial.initialize(device);
-    mPipelineImage.initialize(device);
+    fillShape.initialize(device);
+    fillStroke.initialize(device);
+    solid.initialize(device);
+    linear.initialize(device);
+    radial.initialize(device);
+    image.initialize(device);
+    blit.initialize(device);
+    blitColor.initialize(device);
+    // composition pipelines
+    compAlphaMask.initialize(device, cShaderSource_PipelineCompAlphaMask);
+    compInvAlphaMask.initialize(device, cShaderSource_PipelineCompInvAlphaMask);
+    compLumaMask.initialize(device, cShaderSource_PipelineCompLumaMask);
+    compInvLumaMask.initialize(device, cShaderSource_PipelineCompInvLumaMask);
+    compAddMask.initialize(device, cShaderSource_PipelineCompAddMask);
+    compSubtractMask.initialize(device, cShaderSource_PipelineCompSubtractMask);
+    compIntersectMask.initialize(device, cShaderSource_PipelineCompIntersectMask);
+    compDifferenceMask.initialize(device, cShaderSource_PipelineCompDifferenceMask);
 }
 
 
 void WgPipelines::release()
 {
-    WgBindGroupCompose::releaseLayout();
+    WgBindGroupBlit::releaseLayout();
     WgBindGroupPicture::releaseLayout();
     WgBindGroupRadialGradient::releaseLayout();
     WgBindGroupLinearGradient::releaseLayout();
     WgBindGroupSolidColor::releaseLayout();
     WgBindGroupCanvas::releaseLayout();
-    mPipelineImage.release();
-    mPipelineRadial.release();
-    mPipelineLinear.release();
-    mPipelineSolid.release();
-    mPipelineFillStroke.release();
-    mPipelineFillShape.release();
-}
\ No newline at end of file
+    compDifferenceMask.release();
+    compIntersectMask.release();
+    compSubtractMask.release();
+    compAddMask.release();
+    compInvLumaMask.release();
+    compLumaMask.release();
+    compInvAlphaMask.release();
+    compAlphaMask.release();
+    blitColor.release();
+    blit.release();
+    image.release();
+    radial.release();
+    linear.release();
+    solid.release();
+    fillStroke.release();
+    fillShape.release();
+}
+
+
+WgPipelineComposition* WgPipelines::getCompositionPipeline(CompositeMethod method)
+{
+    switch (method) {
+        case CompositeMethod::ClipPath:
```

Comment:
Alpha blending does not take into account the ClipPath. it's more like a scissor behavior.